module/VuFind/src/VuFind/Controller/Plugin/AbstractRequestBase.php

```diff
@@ -136,13 +136,16 @@ public function rememberValidId($id)
     }
 
     /**
-     * Get remembered valid IDs
+     * Validate supplied IDs against remembered IDs. Returns true if all supplied
+     * IDs are remembered, otherwise returns false.
      *
-     * @return array
+     * @param array $ids IDs to validate
+     *
+     * @return bool
      */
-    public function getValidIds(): array
+    public function validateIds($ids): bool
     {
-        return $this->getSession()->validIds ?? [];
+        return array_diff($ids, $this->getValidIds());
```

Comment:
Should we typecast this to (bool) or use some other check that creates a bool in order to be a little more strict about our typing?